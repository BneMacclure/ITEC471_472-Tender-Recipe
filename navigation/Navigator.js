import React from 'react'
import { createStackNavigator } from '@react-navigation/stack';
//import { DrawerNavigator } from '@react-navigation';
import { Button, View, TouchableOpacity, StatusBar, Alert } from "react-native";
import Icon from 'react-native-vector-icons/Octicons';
import LoginPage from '../components/LoginScreenInfo'
import MainScreen from '../components/MainScreenInfo'
import MainScreenAlt from '../components/MainScreenAlt'

//you can implement different kinds of navigators here. Stack navigators, tab navigators, etc

const Stack = createStackNavigator()

const MenuIcon = () => <Icon
    name='three-bars'
    size={30}
    color='#fff'
    //onPress={() => navigate('DrawerOpen')}
/>;

const Navigator = props => {
    return (
        <Stack.Navigator>
            <Stack.Screen
                name="Main Screen"
                component={MainScreen}
                options={{
                    title: "Main Page",
                    headerStyle: {
                        backgroundColor: '#e35514',
                    },
                    headerTintColor: '#f7f5f2',
                    headerTitleStyle: {
                        fontWeight: 'bold',
                        fontSize: 25,
                    },
                    headerRight: () => (
                        <View
                            style={{
                                flex: 1,
                                alignItems: 'center',
                                flexDirection: 'row',
                                paddingHorizontal: 10,
                                height: StatusBar.currentHeight,
                            }}>
                            <Icon name="three-bars" size={30} color={'#fff'} />
                        </View>
                        ),
                        
                }}
            />
            <Stack.Screen
                name="Login Page"
                component={LoginPage}
                //options={(route)} => ({title: route.params.name})
            />
        </Stack.Navigator>
    );
}

export default Navigator