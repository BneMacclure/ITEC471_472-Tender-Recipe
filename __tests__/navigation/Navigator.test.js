import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Navigator from '../../navigation/Navigator';

describe('<Navigator />', () => {
    it('Navigator snapshot test' , () => {
        const tree = renderer.create(<Navigator />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})