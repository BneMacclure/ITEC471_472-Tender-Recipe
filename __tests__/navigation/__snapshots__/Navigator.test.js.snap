// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigator /> Navigator snapshot test 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#e35514",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#f7f5f2",
                    "fontSize": 25,
                    "fontWeight": "bold",
                  }
                }
              >
                Welcome!
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        accessibilityIgnoresInvertColors={true}
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <Image
                          resizeMode="cover"
                          source={1}
                          style={
                            Array [
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              },
                              Object {
                                "height": undefined,
                                "width": undefined,
                              },
                              Object {},
                            ]
                          }
                        />
                        <Image
                          resizeMode="contain"
                          source={1}
                          style={
                            Object {
                              "alignSelf": "center",
                              "borderColor": "rgba(255,255,255,1)",
                              "borderRadius": 57,
                              "borderWidth": 4,
                              "height": 200,
                              "marginTop": 88.93333333333334,
                              "width": 200,
                            }
                          }
                        />
                        <Text
                          style={
                            Object {
                              "alignSelf": "center",
                              "color": "rgba(251,251,251,1)",
                              "fontFamily": "BigShouldersDisplay_700Bold",
                              "fontSize": 36,
                              "marginTop": 10,
                            }
                          }
                        >
                          Tender Recipes
                        </Text>
                        <View
                          style={
                            Object {
                              "alignSelf": "center",
                              "flexDirection": "row",
                              "height": 35,
                              "marginRight": 30,
                              "marginTop": 60,
                              "width": 276,
                            }
                          }
                        >
                          <Text />
                          <TextInput
                            allowFontScaling={true}
                            inlineImagePadding={0}
                            onChangeText={[Function]}
                            placeholder="Email"
                            placeholderTextColor="rgba(230, 230, 230,1)"
                            rejectResponderTermination={true}
                            selectionColor="rgba(52,250,215,1)"
                            style={
                              Object {
                                "borderColor": "rgba(255,255,255,1)",
                                "borderRadius": 10,
                                "borderWidth": 3,
                                "color": "rgba(255,255,255,1)",
                                "fontSize": 16,
                                "height": 35,
                                "letterSpacing": 1,
                                "lineHeight": 14,
                                "marginLeft": 4,
                                "paddingLeft": 8,
                                "textAlign": "left",
                                "width": 275,
                              }
                            }
                            testID="emailInput"
                            underlineColorAndroid="transparent"
                          />
                        </View>
                        <View
                          style={
                            Object {
                              "alignSelf": "center",
                              "flexDirection": "row",
                              "height": 35,
                              "marginRight": 30,
                              "marginTop": 26,
                              "width": 276,
                            }
                          }
                        >
                          <Text />
                          <TextInput
                            allowFontScaling={true}
                            onChangeText={[Function]}
                            placeholder="Password"
                            placeholderTextColor="rgba(230, 230, 230,1)"
                            rejectResponderTermination={true}
                            secureTextEntry={true}
                            selectionColor="rgba(52,250,215,1)"
                            style={
                              Object {
                                "borderColor": "rgba(255,255,255,1)",
                                "borderRadius": 10,
                                "borderWidth": 3,
                                "color": "rgba(255,255,255,1)",
                                "fontSize": 16,
                                "height": 35,
                                "letterSpacing": 1,
                                "paddingLeft": 8,
                                "width": 275,
                              }
                            }
                            testID="passwordInput"
                            underlineColorAndroid="transparent"
                          />
                        </View>
                        <View
                          style={
                            Object {
                              "alignSelf": "center",
                              "elevation": 5,
                              "flexDirection": "row",
                              "height": 36,
                              "marginTop": 30,
                              "shadowColor": "rgba(206,203,203,1)",
                              "shadowOffset": Object {
                                "height": 3,
                                "width": 3,
                              },
                              "shadowOpacity": 1,
                              "shadowRadius": 0,
                            }
                          }
                        >
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "backgroundColor": "rgba(255,255,255,1)",
                                "borderRadius": 10,
                                "height": 36,
                                "justifyContent": "center",
                                "opacity": 1,
                                "width": 149,
                              }
                            }
                            testID="loginButton"
                          >
                            <Text
                              style={
                                Object {
                                  "alignSelf": "center",
                                  "color": "rgba(249,71,35,1)",
                                }
                              }
                            >
                              Login
                            </Text>
                          </View>
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "backgroundColor": "rgba(255,255,255,1)",
                                "borderRadius": 10,
                                "height": 36,
                                "justifyContent": "center",
                                "marginLeft": 4,
                                "opacity": 1,
                                "width": 149,
                              }
                            }
                            testID="registerButton"
                          >
                            <Text
                              style={
                                Object {
                                  "alignSelf": "center",
                                  "color": "rgba(249,71,35,1)",
                                }
                              }
                            >
                              Sign up
                            </Text>
                          </View>
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "backgroundColor": "rgba(255,255,255,1)",
                                "borderRadius": 10,
                                "height": 36,
                                "justifyContent": "center",
                                "marginLeft": 4,
                                "opacity": 1,
                                "width": 149,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "alignSelf": "center",
                                  "color": "rgba(249,71,35,1)",
                                }
                              }
                            >
                              Debug Login
                            </Text>
                          </View>
                        </View>
                        <ActivityIndicator
                          animating={false}
                          color="rgba(254,242,94,1)"
                          hidesWhenStopped={true}
                          size="large"
                          style={
                            Object {
                              "alignSelf": "center",
                              "marginTop": 50,
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
